src/pytorch_lightning/core/module.py

```diff
@@ -1792,7 +1792,7 @@ def _verify_is_manual_optimization(self, fn_name: str) -> None:
             )
 
     @classmethod
-    def _auto_collect_arguments(cls, frame=None) -> Tuple[Dict, Dict]:  # type: ignore[no-untyped-def]
+    def _auto_collect_arguments(cls, frame: Optional[types.FrameType] = None) -> Tuple[Dict, Dict]:
```

Comment:
@awaelchli this caused different errors

```sh
rc/pytorch_lightning/core/module.py:1811: error: Item "None" of "Optional[FrameType]" has no attribute "f_back"  [union-attr]
src/pytorch_lightning/core/module.py:1811: error: Argument 1 to "collect_init_args" has incompatible type "Union[FrameType, None, Any]"; expected "FrameType"  [arg-type]
```

Reply:
@awaelchli I updated in https://github.com/Lightning-AI/lightning/pull/13603/commits/049ace94f1fb7e3137c95d4e82c10e15e249b8e2

I had to add (at ln1810):

```python
frame_args = collect_init_args(frame.f_back, [])  # type: ignore[union-attr, arg-type]
```

because updating `pytorch_lightning.utilities.parsing.collect_init_args` to accept an Optional[types.FrameType] caused more mypy errors than it resolved.